File: miraculous-frontend/miraculous-ui/src/dto/user-profile-dto/user-profile-dto.ts
export class UserProfileDto {

  id: number;
  login: string;
  email: string;
  dateOfRegistration: string;
  description: string;
  location: string;
  pictureUrlName: string;
  isLikedMusicShown: boolean;


  constructor(id?: number, login?: string, email?: string, dateOfRegistration?: string,
              description?: string, location?: string, pictureUrl?: string, isLikedMusicShown?: boolean) {
    this.id = id;
    this.login = login;
    this.email = email;
    this.dateOfRegistration = dateOfRegistration;
    this.description = description;
    this.location = location;
    this.pictureUrlName = pictureUrl;
    this.isLikedMusicShown = isLikedMusicShown;
  }
}

File: miraculous-backend/src/main/java/com/example/miraculousbackend/dto/UserProfileDto.java
package com.example.miraculousbackend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserProfileDto {

    private Long id;
    private String login;
    private String email;
    private String dateOfRegistration;
    private List<String> favouriteGenres;
    private String description;
    private String location;
    private String pictureUrlName;
    private Boolean isLikedMusicShown;

}
